File: codes/python/p056.py
# 
# Solution to Project Euler problem 56
# Copyright (c) Sankha Saha. All rights reserved.
# 
#  
#   
# 


def compute():
	ans = max(sum(int(c) for c in str(a**b))
		for a in range(100) for b in range(100))
	return str(ans)


if __name__ == "__main__":
	print(compute())

File: codes/Haskell/p056.hs
{- 
 - Solution to Project Euler problem 56
 - Copyright (c) Sankha Saha. All rights reserved.
 - 
 -  
 -   
 -}

import qualified EulerLib


main = putStrLn (show ans)
ans = foldl1 max [EulerLib.digitSum (a^b) | a <- [0..99], b <- [0..99]]
